### PR TITLE
Move CodeStyle layer to Microsoft.CodeAnalysis 3.6.0

### PR DESCRIPTION
**Pull request by mavasani**
Also remove bunch of conditional code that was added due to older version of Microsoft.CodeAnalysis in CodeStyle layer

### CODE REVIEW

File: src/Analyzers/CSharp/CodeFixes/UseDeconstruction/CSharpUseDeconstructionCodeFixProvider.cs [hunk 1]
```diff
@@ -120,10 +120,7 @@ private ForEachVariableStatementSyntax CreateForEachVariableStatement(INamedType
             // However, convert the existing declaration over to a "var (x, y)" declaration or (int x, int y)
             // tuple expression.
             return SyntaxFactory.ForEachVariableStatement(
-#if !CODE_STYLE // ForEachStatementSyntax.AttributeLists is not available in the version of Microsoft.CodeAnalysis used by CodeStyle layer
-                // https://github.com/dotnet/roslyn/issues/41462#issuecomment-595893953 tracks removing these conditional directives.
                 forEachStatement.AttributeLists,
-#endif
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/CodeFixes/UseDeconstruction/CSharpUseDeconstructionCodeFixProvider.cs:126]: nice.  how are you finding this.  are you just going to all CODE_STYLE locations and trying to remove them?

File: src/Analyzers/Core/Analyzers/PopulateSwitch/AbstractPopulateSwitchDiagnosticAnalyzer.cs [hunk 2]
```diff
@@ -57,11 +57,14 @@ private void AnalyzeOperation(OperationAnalysisContext context)
                     .Add(PopulateSwitchStatementHelpers.MissingCases, missingCases.ToString())
                     .Add(PopulateSwitchStatementHelpers.MissingDefaultCase, missingDefaultCase.ToString());
 
+#pragma warning disable CS8620 // Mismatch in nullability of 'properties' parameter and argument types - Parameter type for 'properties' has been updated to 'ImmutableDictionary<string, string?>?' in newer version of Microsoft.CodeAnalysis (3.7.x).
                 var diagnostic = Diagnostic.Create(
                     Descriptor,
                     GetDiagnosticLocation(switchBlock),
                     properties: properties,
                     additionalLocations: new[] { switchBlock.GetLocation() });
+#pragma warning restore CS8620
```
Comment (CyrusNajmabadi) [inline src/Analyzers/Core/Analyzers/PopulateSwitch/AbstractPopulateSwitchDiagnosticAnalyzer.cs:66]: note certain i get what's going on. the local `properties` is non-null... so why is it a problem to pass that along.

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/LanguageServices/TypeInferenceService/AbstractTypeInferenceService.AbstractTypeInferrer.cs [hunk 3]
```diff
@@ -79,11 +79,7 @@ private ImmutableArray<TypeInferenceInfo> Filter(IEnumerable<TypeInferenceInfo>
             }
 
             protected IEnumerable<TypeInferenceInfo> CreateResult(SpecialType type, NullableAnnotation nullableAnnotation = NullableAnnotation.None)
-                => CreateResult(Compilation.GetSpecialType(type)
-#if !CODE_STYLE // TODO: remove this #if directive once the below public API is available in CodeStyle layer.
-                        .WithNullableAnnotation(nullableAnnotation)
-#endif
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/LanguageServices/TypeInferenceService/AbstractTypeInferenceService.AbstractTypeInferrer.cs:85]: does these changes impact any tests anywhere?

File: src/Analyzers/CSharp/CodeFixes/UseDeconstruction/CSharpUseDeconstructionCodeFixProvider.cs (hunk 1 above)
Comment (mavasani) [reply src/Analyzers/CSharp/CodeFixes/UseDeconstruction/CSharpUseDeconstructionCodeFixProvider.cs:126]: Yep, exactly.

File: src/Analyzers/Core/Analyzers/PopulateSwitch/AbstractPopulateSwitchDiagnosticAnalyzer.cs (hunk 2 above)
Comment (mavasani) [reply src/Analyzers/Core/Analyzers/PopulateSwitch/AbstractPopulateSwitchDiagnosticAnalyzer.cs:66]: So, the signature of `Diagnostic.Create` was changed post 3.6.0 in https://github.com/dotnet/roslyn/pull/43023/files#diff-b3066efb626184718718c2d1642f1266R109, but the CodeStyle layer still does not reference the package version that has that change, hence the diagnostic. Once we bump up the CodeStyle layer's MS.CA version to 3.7.0, we should be able to revert this suppression.

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/LanguageServices/TypeInferenceService/AbstractTypeInferenceService.AbstractTypeInferrer.cs (hunk 3 above)
Comment (mavasani) [reply src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/LanguageServices/TypeInferenceService/AbstractTypeInferenceService.AbstractTypeInferrer.cs:85]: Good question - I did also scan through the conditional directives in tests, but let me double check if something specific to this feature had tests disabled in CodeStyle layer.

Comment (mavasani) [reply src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/LanguageServices/TypeInferenceService/AbstractTypeInferenceService.AbstractTypeInferrer.cs:85]: You are indeed correct - good catch! I did miss unskipping tests that were failing due to this. Thanks!

**Pull request merged**